src/Sarif/Writers/SarifLogger.cs

```diff
@@ -479,6 +479,16 @@ public void LogToolNotification(Notification notification)
             _run.Invocations[0].ExecutionSuccessful &= notification.Level != FailureLevel.Error;
 
             CaptureFilesInNotification(notification);
+
+            if (associatedRule != null)
```

Comment:
>associatedRule [](http://example.com/codeflow?start=16&length=14)

With this change, the rule metadata for any notification associated with a rule will be persisted to the rules table in the log.